plugins/ingest-attachment/build.gradle

```diff
@@ -71,9 +71,9 @@ dependencies {
   api "org.apache.pdfbox:fontbox:${versions.pdfbox}"
   api "org.apache.pdfbox:jempbox:1.8.17"
   api "commons-logging:commons-logging:${versions.commonslogging}"
-  api "org.bouncycastle:bcmail-jdk15to18:${versions.bouncycastle}"
-  api "org.bouncycastle:bcprov-jdk15to18:${versions.bouncycastle}"
-  api "org.bouncycastle:bcpkix-jdk15to18:${versions.bouncycastle}"
+  api "org.bouncycastle:bcmail-jdk18on:${versions.bouncycastle}"
```

Comment:
In main we support java three different versions of the jdk [1], only of them seems to align with jdk18on.

- [1] Supported JDK matrix 
https://github.com/opensearch-project/OpenSearch/blob/84750b3840b8249ac1363e2e1bfa6b24c174e15f/.github/workflows/assemble.yml#L10

Reply:
jdk18on supports all java versions from 8+ onwards. The naming convention of BouncyCastle is simply confusing since they picked it before Java 15 was released.

Reply:
I think this is fine from JDK version perspective, the BC is used only in plugins so should not break things

Reply:
> jdk18on -> jdk 1.8 on

🙄 Yup - that is confusing.  Luckily, with that resolved it means we can do the easy thing and pull in this change